### PR TITLE
First draft of a manpage in Markdown format.

### PR DESCRIPTION
**Pull request by tomperrine**
This is a first draft of a man page for the signalbackup-tool.

It has many flaws and bugs, but has re-factored the info from the README into arguments, selectors, etc.

I have not yet dived into the code to get info on the un-documented arguments, etc.

### CODE REVIEW

**Comment by Meteor0id** [comment]
Maybe add a warning / notion that using this tool means unencrypted Signal chats hit your disk, and could leave forensic traces. Signal was after all designed to never leave anything unencrypted, so this tool should be used with the consideration that your chats are less secure when you use this tool.

**Comment by bepaald** [comment]
Note that this program should not cause your data to hit your disk unless:
1. Explicitly asked for it (for example [Dump unencrypted database to disk](https://github.com/bepaald/signalbackup-tools#dump))
2. Your RAM is full (probably from other processes) and things are written to swap. This program shouldn't use a lot of RAM (not like the entire backup is read in there, just piece by piece when needed). I could investigate using [mlockall](https://man7.org/linux/man-pages/man2/mlock.2.html) to prevent this program from ever swapping, but I have no experience with this. I think swapping is pretty rare these days, but on systems that regularly swap due to low RAM, the user may not even appreciate a program using `mlock`.

So, maybe this note is not necessary? I think both of the above conditions are pretty much implied when working with a program like this. On the other hand, it can't hurt.

File: signalbackup-tool.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    $ ./signalbackup-tools [input] [password] --runsqlquery "UPDATE sms SET body = 'No' WHERE _id == 21300" --ouput [output]
+
+If you also need to edit the attachments, dump the backup to directory first (as described above) and do whatever you want, but realize when editing the .bin file, it will usually require changes to also be made to the .sbf file and the sql database to end up with a valid database.
+
+# AUTHOR
+
+https://github.com/bepaald
+
+# REPORTING BUGS
```
Comment (Willdickem) [inline signalbackup-tool.md:319]: It's on my phone

**Review by Willdickem — CHANGES_REQUESTED** [review]

File: signalbackup-tool.md (hunk 1 above)
Comment (Willdickem) [reply signalbackup-tool.md:319]: -

**Pull request closed without merging**